src/middleware.ts

```diff
@@ -122,13 +122,52 @@ type StateStorage = {
   setItem: (name: string, value: string) => void | Promise<void>
 }
 type PersistOptions<S> = {
+  /** Name of the storage (must be unique) */
   name: string
-  storage?: StateStorage
+  /**
+   * A function returning a storage.
+   * The storage must fit `window.localStorage`'s api.
+   * For example the storage could be `AsyncStorage` from React Native.
+   *
+   * @default () => localStorage
+   */
+  storage?: () => StateStorage
+  /**
+   * Use a custom serializer.
+   * The returned string will be stored in the storage.
+   *
+   * @default JSON.stringify
+   */
   serialize?: (state: S) => string | Promise<string>
+  /**
+   * Use a custom deserializer.
+   * The returned value will be used as the new state.
+   *
+   * @param str The storage's current value.
+   * @default JSON.parse
+   */
   deserialize?: (str: string) => S | Promise<S>
+  /**
+   * Prevent some items from being stored.
+   * Nested properties can be accessed like so: `foo.bar.baz`
```

Comment:
Nested properties are not supported, right?

Reply:
Yes, I planned to add it but forgot to remove this comment